src/verify.cc

```diff
@@ -35,16 +35,21 @@ inline const uint8_t* castToUChar(const absl::string_view& str) {
   return reinterpret_cast<const uint8_t*>(str.data());
 }
 
-bool verifySignatureRSA(EVP_PKEY* key, const EVP_MD* md,
-                        const uint8_t* signature, size_t signature_len,
-                        const uint8_t* signed_data, size_t signed_data_len) {
+bool verifySignatureRSA(RSA* key, const EVP_MD* md, const uint8_t* signature,
+                        size_t signature_len, const uint8_t* signed_data,
+                        size_t signed_data_len) {
   if (key == nullptr || md == nullptr || signature == nullptr ||
       signed_data == nullptr) {
     return false;
   }
-  bssl::UniquePtr<EVP_MD_CTX> md_ctx(EVP_MD_CTX_create());
+  bssl::UniquePtr<EVP_PKEY> evp_pkey(EVP_PKEY_new());
+  if (EVP_PKEY_set1_RSA(evp_pkey.get(), key) != 1) {
```

Comment:
How slow is this EVP_PKey_set1_RSA call?  We are doing this for every JWT verification.  

In jwks parsing: (in envoy jwt_authn filter, this is done once for each jwks)
* RSA format:  generate RSA directly,
* pkcs and x509,   generate Pkey, and convert to RSA, 

in every verify() call,  we convert RSA back to Pkey.

I feel for better performance, we should keep Pkey format in jwks parsing.

Reply:
I did a quick perf profile by running a jwks_rsa test 100 times in a loop.  The conversion took 0.002% of the run time, so I don't think the performance of the conversion is an issue.